api/tests/integration/infrastructure/repositories/certification-course-repository_test.js

```diff
@@ -84,14 +85,73 @@ describe('Integration | Repository | Certification Course', function() {
         expect(thisCertificationCourse.isPublished).to.be.ok;
       });
 
-      it('should retrieve associated assessment and challenges with the certification course', async () => {
+      it('should retrieve associated challenges with the certification course', async () => {
         // when
         const thisCertificationCourse = await certificationCourseRepository.get(courseId);
 
         // then
-        expect(thisCertificationCourse.assessment.id).to.equal(assessmentId);
         expect(thisCertificationCourse.challenges.length).to.equal(2);
       });
+
+      context('When the certification course has several assessments', () => {
+
+        context('When one of those assessment is completed', () => {
+          let completedAssessmentId;
+
+          beforeEach(() => {
+            databaseBuilder.factory.buildAssessment({ courseId: courseId, userId, state: Assessment.states.STARTED });
+            completedAssessmentId = databaseBuilder.factory.buildAssessment({ courseId: courseId, userId }).id;
+            return databaseBuilder.commit();
+          });
+
+          it('should retrieve associated completed assessment', async () => {
+            // when
+            const thisCertificationCourse = await certificationCourseRepository.get(courseId);
+
+            // then
+            expect(thisCertificationCourse.assessment.id).to.equal(completedAssessmentId);
+          });
+        });
+
+        context('When no assessment is completed', () => {
+
+          beforeEach(() => {
+            databaseBuilder.factory.buildAssessment({ courseId: courseId, userId, state: Assessment.states.STARTED });
+            databaseBuilder.factory.buildAssessment({ courseId: courseId, userId, state: Assessment.states.STARTED });
+            databaseBuilder.factory.buildAssessment({ courseId: courseId, userId, state: Assessment.states.STARTED });
+            return databaseBuilder.commit();
+          });
+
+          it('should retrieve an assessment anyway', async () => {
+            // when
+            const thisCertificationCourse = await certificationCourseRepository.get(courseId);
+
+            // then
+            expect(thisCertificationCourse.assessment.id).to.exist;
```

Comment:
pourquoi ne pas tester que l'id de celui retourné est parmi [assessmentID1, assessmentID2, assessmentID3] ? `to.exist` n'est pas très précis

Reply:
Ok je vais faire ça dans la journée.